clients/java/src/test/java/io/zeebe/client/workflow/CreateWorkflowInstanceWithResultTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.client.workflow;
+
+import static io.zeebe.client.util.JsonUtil.fromJsonAsMap;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.entry;
+
+import io.zeebe.client.api.response.WorkflowInstanceResult;
+import io.zeebe.client.util.ClientTest;
+import io.zeebe.gateway.protocol.GatewayOuterClass.CreateWorkflowInstanceWithResultRequest;
+import java.time.Duration;
+import org.junit.Test;
+
+public class CreateWorkflowInstanceWithResultTest extends ClientTest {
+
+  @Test
+  public void shouldCreateWorkflowInstanceByWorkflowInstanceKey() {
+    // given
+    gatewayService.onCreateWorkflowInstanceWithResultRequest(123, "testProcess", 12, 32);
+
+    // when
+    final WorkflowInstanceResult response =
```

Comment:
The key point are the variables I think we should test them as well. That we are able to receive them from the response as map, as object etc.